### PR TITLE
Add ReCaptcha

### PR DESCRIPTION
**Pull request by caiseL**
imagen

imagen

### CODE REVIEW

File: server/src/test/scala/apis/ReCaptchaApiSpec.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
+package apis
```
Comment (AlexITC) [inline server/src/test/scala/apis/ReCaptchaApiSpec.scala:1]: Let's the same package than the one used by the class being tested, `net.wiringbits.apis`

File: web/src/main/scala/net/wiringbits/components/widgets/ReCaptcha.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package net.wiringbits.components.widgets
+
+import net.wiringbits.common.models.Captcha
+import slinky.core.FunctionalComponent
+import slinky.core.annotations.react
+import typings.reactGoogleRecaptcha.components.ReactGoogleRecaptcha
+
+@react object ReCaptcha {
+  case class Props(onChange: Option[Captcha] => Unit)
```
Comment (AlexITC) [inline web/src/main/scala/net/wiringbits/components/widgets/ReCaptcha.scala:9]: Let's receive the catpcha key by argument

File: web/src/main/scala/net/wiringbits/components/widgets/ReCaptcha.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  private val siteKey = net.wiringbits.BuildInfo.recaptchaKey.filter(_.nonEmpty).getOrElse {
```
Comment (AlexITC) [inline web/src/main/scala/net/wiringbits/components/widgets/ReCaptcha.scala:11]: This should be propagated from the `Main/App` instead

File: web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala [hunk 4]
```diff
@@ -55,6 +59,7 @@ import scala.util.{Failure, Success}
 
       val email = state.email.getOrElse("")
       val password = state.password.getOrElse("")
+      val captcha = Captcha.trusted(state.captcha.map(_.string).getOrElse(""))
```
Comment (AlexITC) [inline web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala:62]: `trusted` should almost never be invoked outside tests

File: web/src/main/scala/net/wiringbits/components/widgets/SignUpForm.scala [hunk 5]
```diff
@@ -59,6 +63,7 @@ import scala.util.{Failure, Success}
       val name = state.name.getOrElse("")
       val email = state.email.getOrElse("")
       val password = state.password.getOrElse("")
+      val captcha = Captcha.trusted(state.captcha.map(_.string).getOrElse(""))
```
Comment (AlexITC) [inline web/src/main/scala/net/wiringbits/components/widgets/SignUpForm.scala:66]: Same

**Review by AlexITC — CHANGES_REQUESTED** [review]

File: web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala (hunk 4 above)
Comment (caiseL) [reply web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala:62]: I'm gonna fix this when #106 gets accepted

**Review by AlexITC — APPROVED** [review]

File: web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala (hunk 4 above)
Comment (AlexITC) [reply web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala:62]: Merged, thanks

**Review by AlexITC — APPROVED** [review]

File: web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala (hunk 4 above)
Comment (AlexITC) [reply web/src/main/scala/net/wiringbits/components/widgets/SignInForm.scala]: I think you are missing to integrate the captcha with the form

**Pull request merged**